Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -540,6 +544,11 @@ export class ResourceGroup {
         }
     }
 
+    private async waitAndPerformAzureDeployment(armClient: armResource.ResourceManagementClient, deployment: Deployment, retryCount): Promise<void> {
+        await sleepFor(3);
```

Comment:
given this is just two lines without much logic, you can inline this function in the caller. This will bring out recursion also clearly